test/phoenix_profiler/telemetry/collector_test.exs

```diff
@@ -24,9 +24,7 @@ defmodule PhoenixProfiler.TelemetryCollectorTest do
 
       {:ok, _} = TelemetryRegistry.register(name, self())
 
-      fn -> :telemetry.execute([:debug, :me], %{}, %{}) end
-      |> Task.async()
-      |> Task.await()
+      :telemetry.execute([:debug, :me], %{}, %{})
```

Comment:
The nature of this test has changed :) The purpose of this test was to ensure we can collect telemetry from spawned processes like Tasks.

However, I missed the fact that we can only rely on [`$callers` on Elixir v1.8+](https://elixir-lang.org/blog/2019/01/14/elixir-v1-8-0-released/) :(

A few thoughts– 

1) I think `$ancestors` would work in this case. We could collect pids from both ancestors and callers if it exists.

2) We could use process links. `Task.async/1` links the caller to the spawned task but I think that collecting telemetry from all linked processes will have unintended side effects.

3) We require Elixir v1.8+. I think this would not have a huge impact on adoption, but I am open to being told I am wrong. :)

Right now I am leaning towards 3 but let me know what you think. :)

Reply:
Gotcha just reverted this change. IMHO option 2 is a bit more risky than just requiring v1.8+ so option 3 👍🏻